### DISCUSSION TITLE
Welcome to Chlore Discussions!

### DISCUSSION THREAD

**Discussion by trap-representation** [trap-representation/Sasm#1] [Announcements]
Hi there, this is where we discuss about the Chlore programming language.

- Just curious about the language?
- Want to learn more about something specific about Chlore?
- Want to discuss about new features being implemented in the language?
- Want to recommend a compilation target for Ehre?

Whatever it is, feel free to post that here.

**Comment by Aero57** [comment]
What will Chlore be used for? What can I do with it? Also, why chlorine, why not Aurum? Also, why's your implementation called MChlore?

**Reply by trap-representation** [reply]
> What will Chlore be used for?

Take a look at the `What Chlore is and isn't` section in the Chlore wiki.

> What can I do with it?

Anything that you can do with "modern" programming languages.

> Also, why chlorine, why not Aurum?

chlorine::=Chlore

It just happened to be one of the first names that had popped into my head.

> Also, why's your implementation called MChlore?

There's no "this is the way you name your implementation" rule for Chlore. You can name your implementation whatever you want. And just in case you're wondering, MChlore stands for Minimal Chlore.

Hope this answers your questions. If you've got any more questions to ask, feel free to create a new discussion. That way it is a bit easier to look them up later.

**Reply by Aero57** [reply]
What do you mean by a "modern programming language"? Also, if this is can be used to do anything a "modern PL" can do, then why does its syntax look so ASM/fortran/C like? Python has better syntax IMO. Also, the wiki says that Chlore is turning into a general purpose language, then why not have the syntax be "general purpose" too? Also, in the wiki, you sometimes say it is a compilation target and other times say it is meant to be a GP PL. So what exactly is it?

**Reply by trap-representation** [reply]
> What do you mean by "modern programming language"?

Languages that aim to *only* target "modern" hardware and have their formal specification written in such a way that if someone wants to write a standard-conforming compiler/interpreter for an exotic architecture, they'll have to emulate almost everything in the language. Example: Languages that mandate 2's complement representation for signed integers.

> Also, if this can be used [...] like?

I really doubt it looks anything like Fortran/C. It definitely looks a bit like Forth (or basically any other stack based language in existance) though.

> Python has better syntax IMO

Cool.

> Also, the wiki says that Chlore [...] "general purpose" too?

I'm not sure what you mean by this. If you think the syntax can be improved, feel free to suggest a change.

> Also, in the wiki, you [...] GP PL. So what exactly is it?"

A general purpose programming language can be a compilation target too.
